### ISSUE TITLE
Type synonym error on Identity: Type synonym ... is partially applied.

### ISSUE THREAD

**Issue by jim108dev** [purescript/purescript#4105]
## Description

I wanted to update my project from PureScript 0.13 to 0.14. Getting the error message:

```txt
Type synonym Misc.Template is partially applied.
Type synonyms must be applied to all of their type arguments.
```

## To Reproduce

Misc.purs:

```purescript
module Misc where
  
type Template col = { bio :: col String }

type Identity a = a

type Patch = Template Identity
```

+ Other.purs:
```purescript
module Other where

import Prelude

import Misc 

newtype UpdateDto = UpdateDto Patch

derive instance eqUpdateDto :: Eq UpdateDto
```
- It can’t be triggered in one file.
- It has one of those error positions where it’s caused by the existence of Other.purs, but identified in Misc.purs
- It can be worked around by inlining the Identity synonym usage (newtype UpdateDto = UpdateDto (Template Identity))
- Possibly related to  #4101.

## Expected behavior

No error like in version 0.13.

## Additional context

[discourse](https://discourse.purescript.org/t/update-purescript-0-13-0-14-type-synonyms/2373)
## PureScript version

0.14.2

**Comment by hdgarrood** [comment]
This definitely seems like a bug. Thanks for the report!

**Comment by MonoidMusician** [comment]
Is there any way to debug the externs now that they are encoded in CBOR? I ran `cbor2diag.rb output/Misc/externs.cbor` but it's hard to interpret the output: https://pastebin.com/VcPcgE5L

It seems like something is going wrong with polykinds in particular: giving the kind signature `type Identity :: Type -> Type` also results in no error.

**Comment by paulyoung** [comment]
I think I may be running into this when trying to use some old code that uses the `RowApply` operator as `Type.Row (type (+))` with a recent version of the compiler.

e.g. https://github.com/paulyoung/purescript-style/blob/cf17c7d7c50dd8f6a7060f399fd76f4b596ff492/src/Style/Declaration/Value.purs#L16

```
[1/1 PartiallyAppliedSynonym] (unknown module)

  Type synonym Type.Row.RowApply is partially applied.
  Type synonyms must be applied to all of their type arguments
```

I'm doing this as a workaround:

```purescript
type RowApply (f :: Row Type -> Row Type) (a :: Row Type) = f a

infixr 0 type RowApply as +
```

**Comment by rhendric** [comment]
That very much looks like this, yes.

Just FYI, I have an implementation of #4086 that fixes this issue (among others) sitting on my hard drive, slightly musty with bit-rot. I ended up on a bit of a yak shave last year trying to make #4169 less ad-hoc, but once I've finished reacting to 0.15-related work (I'm currently bringing #3915 up to date), #4169 and then #4086 are next on my personal non-breaking to-do list, which will take care of this issue. No guarantees about a timeline for a merge but know that I haven't forgotten about self-assigning this!
